pkg/lib/credentials.js

```diff
@@ -288,10 +295,16 @@
             return dfd.promise();
         };
 
-        self.unload = function unload(name) {
-            return cockpit.spawn(["ssh-add", "-d", name],
-                    { pty: true, err: "message", directory: self.path })
-                .done(refresh);
+        self.unload = function unload(key) {
+            var proc;
+            var options = { pty: true, err: "message", directory: self.path };
+
+            if (key.name && !key.agent_only)
```

Comment:
What's the difference between agent_only and !agent_only? Why doesn't everything go via the agent?

Reply:
We only know about the key because it's in the agent not in the users home dir. These are keys that we might not have corresponding .pub file for so we remove them from the agent differently.